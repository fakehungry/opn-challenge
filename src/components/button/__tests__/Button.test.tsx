import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '..';

describe('button component', () => {
  it('should render button with text and match previous snapshot', () => {
    const { container } = render(<Button className="test-btn">Button</Button>);

    expect(container).toMatchSnapshot();
  });

  it('should change background color on hover', () => {
    const { container } = render(<Button>Button</Button>);

    fireEvent.mouseEnter(container);
    expect(container).toHaveStyle('background-color: #386bee, color: white');

    fireEvent.mouseLeave(container);
    expect(container).toHaveStyle(
      'background-color: transparent, color: #386bee',
    );
  });

  it('should render button with text and onClick event', () => {
    const mockFn = jest.fn();
    const { getByText } = render(<Button onClick={mockFn}>Button</Button>);

    getByText('Button').click();
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
