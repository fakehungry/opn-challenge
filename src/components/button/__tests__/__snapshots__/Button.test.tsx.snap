// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should render button with text and match previous snapshot 1`] = `
<div>
  <button
    class="sc-beySPh iLXLnS test-btn"
  >
    Button
  </button>
</div>
`;
